ontology/yaml/resources/HVAC/entity_types/TK.yaml

```diff
@@ -0,0 +1,30 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the License);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an AS IS BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+########################
+### Canonical Types  ###
+########################
+
+TK_TPKD:
+  description: "Tpkd water tank."
```

Comment:
Is this a single tank with three sets of alarms at different positions in the tank? If this were three separate tanks they should be modeled independently.

If this really is one tank, it is probably good to create an abstract type for this. How about something like this:
```
WLAM:
  description: "Water level alarm monitoring, generally for a water tank."
  is_abstract: true
  uses:
  - water_high_level_alarm
  - water_low_level_alarm
  - water_overflow_level_alarm

WLAM3X:
  description: "Water level alarm monitoring, generally for a water tank with three independent sets of alarms."
  is_abstract: true
  uses:
  - water_high_level_alarm_1
  - water_high_level_alarm_2
  - water_high_level_alarm_3
  - water_low_level_alarm_1
  - water_low_level_alarm_2
  - water_low_level_alarm_3
  - water_overflow_level_alarm_1
  - water_overflow_level_alarm_2
  - water_overflow_level_alarm_3

TK_WLAM3X:
  description: "A water tank with three sets of alarms for high, low, and overflow alerting."
  implements:
  - TK
  - WLAM3X
 ```

Reply:
The abstract type has been created